app/api/user/addcomment/route.ts

```diff
@@ -0,0 +1,618 @@
+import { NextResponse } from "next/server";
+import { conn } from "@/services/mongo.service";
+import { findById, findOneAndUpdate, findOne } from "@/utils/user/user";
+import mongoose from "mongoose";
+import { Feed } from "@/utils/models/feed.model";
+
+export async function POST(request: Request) {
+  try {
+    await conn();
+    const req = await request.json();
+    var id = new mongoose.Types.ObjectId();
+    const { user_data_id, content, comment, replyTo } = req;
+    const { success, user, error } = await findOne({ _id: user_data_id });
+    if (!success) {
+      return NextResponse.json({ status: "error", message: error });
+    }
+    const cuser = await findOne({ _id: req.user_id });
+    if (!cuser.success) {
+      return NextResponse.json({ status: "error", message: cuser.user.error });
+    }
+    let announcementData = {
+      announcement: `${cuser.user.username} commented on your post`,
+      post_image: cuser.user.profile_image ? cuser.user.profile_image : null,
+      post_video: null,
+      link: `/homescreen/profile/${cuser.user.username}`,
+      time: Date.now() / 1000,
+      username: cuser.user.username,
+      linkpreview_data: null,
+    };
+
+    if (content.videoId) {
+      let obj = user.videos.find(async (video: any, i: any) => {
```

Comment:
@Sahil412000 Where is this getting redefined, or where is the value getting changed? This should be const if the value of variable `obj` is not getting changed.